shogun-lib/src/main/java/de/terrestris/shogun/lib/service/BaseService.java

```diff
@@ -72,18 +76,41 @@
     @Lazy
     protected UserProviderService userProviderService;
 
+    @Autowired
+    protected List<BaseEntityPermissionEvaluator<?>> permissionEvaluators;
+
     @PostFilter("hasRole('ROLE_ADMIN') or hasPermission(filterObject, 'READ')")
     @Transactional(readOnly = true)
     public List<S> findAll() {
         return (List<S>) repository.findAll();
     }
 
+    @Transactional(readOnly = true)
+    public Page<S> findAll(Pageable pageable) {
+        // note: security check is done in permission evaluator
+        Optional<User> userOpt = userProviderService.getUserBySession();
+
+        // todo: can this be simplified? autowiring BaseEntityPermissionEvaluator did not work.
+        BaseEntityPermissionEvaluator entityPermissionEvaluator =
+            this.getPermissionEvaluatorForClass(BaseEntity.class.getCanonicalName());
+
+        Class<? extends BaseEntity> entityClass = this.getBaseEntityClass();
+
+        return entityPermissionEvaluator.findAll(userOpt.orElseThrow(), pageable, repository, entityClass);
```

Comment:
Do we really want to throw if no user is available? Project implementations might allow anonymous access e.g. to the `/applications` interface where this could fail. Maybe it's safer to provide the check inside the evaluator (which might needed to be implemented in such scenario anyway)?

Reply:
moved the check into the default implementation in fixed in 63741cd4c7a03a279856fdbdf9e73df0cd855a95